geoapps/processing/peak_finder.py

```diff
@@ -2014,9 +2030,19 @@ def get_line_indices(self, line_id):
         """
         Find the vertices for a given line ID
         """
-        indices = np.where(
-            np.asarray(self.survey.get_data(self.lines.data.value)[0].values) == line_id
-        )[0]
+        line_data = self.survey.get_data(self.lines.data.value)[0]
+
+        if isinstance(line_data, ReferencedData):
+            line_id = [
+                key
+                for key, value in line_data.value_map.map.items()
+                if value == line_id
+            ]
```

Comment:
**suggestion:** extract as a function.

same logic repeated in em1d_inversion.py and plotting.py

Maybe a member function on `ReferencedData`?